apps/prairielearn/src/pages/instructorAssessmentAccessOverrides/instructorAssessmentAccessOverrides.ejs

```diff
@@ -0,0 +1,321 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <%- include('../partials/head'); %>
+</head>
+<body>
+  <script>
+    function toDatetimeLocal(date) {
+        var year = date.getFullYear();
+        var month = ('0' + (date.getMonth() + 1)).slice(-2);
+        var day = ('0' + date.getDate()).slice(-2);
+        var hours = ('0' + date.getHours()).slice(-2);
+        var minutes = ('0' + date.getMinutes()).slice(-2);
+        return year + '-' + month + '-' + day + 'T' + hours + ':' + minutes;
+    }
+
+   
+    $(function () {
+      $('[data-toggle="popover"]').popover({ sanitize: false,   container: 'body' });
+
+      $('#add_new_override').on('submit', function (event) {
+        var startDate = new Date($('#start_date').val());
+        var endDate = new Date($('#end_date').val());
+      
+        if (startDate >= endDate) {
+          event.preventDefault();
+          $('#end_date').addClass('is-invalid');
+          $('#end_date_error').text('End date should be greater than the start date.').show();
+        }
+        
+        
+      });
+      $('.edit-override-button').on('click', function (event) {
+        event.preventDefault();
+        var button = $(this);
+        var editOverrideForm = $('#edit-override-form');
+        editOverrideForm.find('#edit-assessment_id').val(button.data('assessment-id'));
+        editOverrideForm.find('#edit-student_uid').val(button.data('user-id'));
+        editOverrideForm.find('#edit-group_id').val(button.data('group-id'));
+        editOverrideForm.find('#edit-credit').val(button.data('credit'));
+        var policy_start_date = new Date(button.data('start-date'));     
+        var policy_start_date_string = toDatetimeLocal(policy_start_date);
+        var policy_end_date = new Date(button.data('end-date'));
+        var policy_end_date_string = toDatetimeLocal(policy_end_date);
```

Comment:
This fails in Safari; did you test in browsers besides Chrome?

Specifically, the string format that you're passing to `new Date` isn't parseable. You can prove this to yourself by popping open a Safari console and running `new Date('2024-10-10 22:00:00 (CDT)')`:

![Screenshot 2024-05-07 at 14 47 37](https://github.com/PrairieLearn/PrairieLearn/assets/1476544/053bef50-9002-430a-9ba6-d5796dbb94b7)

Rather than round-tripping things through a `Date` object on the client, consider doing everything on the server. That is, populate `data-start-date`/`data-end-date` with a string in the format expected by `value` on the `<input type="datetime-local">` (e.g. `2018-06-12T19:30:00`).

Reply:
Ok, after moving to serverside and not using date objects I manually parse the string on the server. let me know what you think [a25ab41](https://github.com/PrairieLearn/PrairieLearn/pull/9103/commits/a25ab41ef1dbde34710e503b47a7eb18a4967f5a)

Reply:
More modifications to this in 
355169b

Reply:
Let's continue discussion in a new thread here: https://github.com/PrairieLearn/PrairieLearn/pull/9103#discussion_r1614074243